app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -511,4 +516,58 @@ public Snackbar showSnackbarAbovePlayer(CharSequence text, int duration) {
     public Snackbar showSnackbarAbovePlayer(int text, int duration) {
         return showSnackbarAbovePlayer(getResources().getText(text), duration);
     }
+
+    //Hardware keyboard support
+    @Override
+    public boolean onKeyUp(int keyCode, KeyEvent event) {
+        AudioManager audioManager = (AudioManager) getSystemService(AUDIO_SERVICE);
+        Integer customKeyCode = null;
+
+        switch (keyCode) {
+            case KeyEvent.KEYCODE_P:
+                //Play/Pause
+                customKeyCode = KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE;
+                break;
+            case KeyEvent.KEYCODE_J: //Fallthrough
+            case KeyEvent.KEYCODE_A:
+            case KeyEvent.KEYCODE_COMMA:
+                //Go Back
+                customKeyCode = KeyEvent.KEYCODE_MEDIA_REWIND;
+                break;
+            case KeyEvent.KEYCODE_K: //Fallthrough
+            case KeyEvent.KEYCODE_D:
+            case KeyEvent.KEYCODE_PERIOD:
+                //Go Forward
+                customKeyCode = KeyEvent.KEYCODE_MEDIA_FAST_FORWARD;
+                break;
+            case KeyEvent.KEYCODE_PLUS: //Fallthrough
+            case KeyEvent.KEYCODE_W:
+                //Raise volume
+                audioManager.adjustStreamVolume(AudioManager.STREAM_MUSIC,
+                        AudioManager.ADJUST_RAISE, AudioManager.FLAG_SHOW_UI);
+                return true;
+            case KeyEvent.KEYCODE_MINUS: //Fallthrough
+            case KeyEvent.KEYCODE_S:
+                //Raise volume
```

Comment:
This should be lowering ;) 

Actually, I would suggest to just remove the comments explaining what is done. I think it's pretty obvious what the code does and it is always just a single line. Comments that basically just repeat the code make the code harder to read (more information readers needs to process).

Comments like `Fallthrough` explain things that developers need to have an eye on. The comment `Go to x% of video` is also fine because when looking at the code, it could take a second to get what it does. In contrast, a comment like `Toggle picture-in-picture mode` immediately followed `compatEnterPictureInPicture();` does not really add value, in my opinion.